plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPropertyPage.java

```diff
@@ -56,81 +103,198 @@ protected Control createContents(Composite parent) {
     createPromoteSection(container);
 
     createAdvancedSection(container);
-    
-    Dialog.applyDialogFont(container); 
-    
+
+    Dialog.applyDialogFont(container);
+
+    loadPreferences();
+
+    setupDataBinding();
+
     return container;
   }
 
-  private void createProjectIdSection(Composite parent) {
-    Composite projectIdComp = new Composite(parent, SWT.NONE);
-    projectIdComp.setLayout(new GridLayout(3, false));
-    projectIdComp.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
-  
-    promptForProjectIdButton = new Button(projectIdComp, SWT.CHECK);
-    promptForProjectIdButton.setText(Messages.getString("deploy.prompt.projectid"));
-    promptForProjectIdButton.setLayoutData(new GridData(SWT.LEFT, SWT.CENTER, true, false, 3, 1));
-    promptForProjectIdButton.addSelectionListener(new SelectionListener() {
-      
+  private void setupDataBinding() {
+    bindingContext = new DataBindingContext();
+
+    setupProjectIdDataBinding(bindingContext);
+    setupProjectVersionDataBinding(bindingContext);
+    setupAutoPromoteDataBinding(bindingContext);
+    setupBucketDataBinding(bindingContext);
+
+    PreferencePageSupport.create(this, bindingContext);
+    bindingContext.updateTargets();
+    observables = new ObservablesManager();
+    observables.addObservablesFromContext(bindingContext, true, true);
+  }
+
+  private void setupProjectIdDataBinding(DataBindingContext dbc) {
+    final ISWTObservableValue overrideObservable =
+        WidgetProperties.selection().observe(overrideProjectIdButton);
+    final ISWTObservableValue projectIdObservable =
+        WidgetProperties.text(SWT.Modify).observe(projectId);
+
+    dbc.addValidationStatusProvider(new MultiValidator(overrideObservable.getRealm()) {
+      @Override
+      protected IStatus validate() {
+        if (overrideObservable.getValue() == Boolean.FALSE) {
```

Comment:
I think the issue was that if someone accidentally creates `new Boolean(b)` (which should be avoided), then `==` or `!=` will give different results than `.equals()` will do.

http://findbugs.sourceforge.net/bugDescriptions.html#RC_REF_COMPARISON_BAD_PRACTICE_BOOLEAN